routers/api/v1/users.go

```diff
@@ -91,20 +94,90 @@ func (r *apiV1Router) Login(ctx *gin.Context) {
 // Headers:  Authorization -> token
 func (r *apiV1Router) Verify(ctx *gin.Context) {
 	token := ctx.GetHeader("Authorization")
+	claims := auth.GetJWTClaims(token, []byte(r.env.Get(environment.JWTSecret)))
+	if claims == nil {
+		models.SendAPIError(ctx, http.StatusUnauthorized, "invalid token")
+		return
+	}
+
+	r.logger.Info("claims", zap.Any("claims", claims))
+	ctx.JSON(http.StatusOK, verifyRes{
+		Response: models.Response{
+			Status: http.StatusOK,
+		},
+	})
+}
 
-	valid := auth.IsValidJWT(token, []byte(r.env.Get(environment.JWTSecret)))
-	if valid {
-		ctx.JSON(http.StatusOK, verifyRes{
-			Response: models.Response{
-				Status: http.StatusOK,
-			},
-		})
-	} else {
-		ctx.JSON(http.StatusUnauthorized, verifyRes{
-			Response: models.Response{
-				Status: http.StatusUnauthorized,
-				Err:    "invalid token",
-			},
-		})
+// GET: /api/v1/users/me
+// Response: status int
+//           error string
+//           user entities.User
+// Headers:  Authorization -> token
+func (r *apiV1Router) GetMe(ctx *gin.Context) {
+	token := ctx.GetHeader("Authorization")
+	claims := auth.GetJWTClaims(token, []byte(r.env.Get(environment.JWTSecret)))
+	if claims == nil {
+		r.logger.Warn("invalid token", zap.String("token", token))
+		models.SendAPIError(ctx, http.StatusUnauthorized, "invalid token")
+		return
+	}
+
+	user, err := r.userService.GetUserWithID(ctx, claims.Id)
+	if err != nil {
+		if err == services.ErrNotFound {
+			r.logger.Warn("user not found", zap.Any("id", claims.Id))
+			models.SendAPIError(ctx, http.StatusBadRequest, "user not found")
+		} else {
+			r.logger.Error("could not fetch user", zap.Error(err))
+			models.SendAPIError(ctx, http.StatusInternalServerError, "there was a problem with fetching the user")
+		}
+		return
 	}
+	ctx.JSON(http.StatusOK, getMeRes{
+		User: *user,
+	})
+}
+
+// PUT: /api/v1/users/me
+// Request:  name string
+//           team primitive.ObjectID.Hex
+// Response: status int
+//           error string
+// Headers:  Authorization -> token
+func (r *apiV1Router) PutMe(ctx *gin.Context) {
+	name := ctx.PostForm("name")
+	team := ctx.PostForm("team")
+
+	if len(name) == 0 && len(team) == 0 {
+		r.logger.Warn("neither name nor team provided")
+		models.SendAPIError(ctx, http.StatusBadRequest, "either name or team must be provided")
```

Comment:
I think this should say "both the name and team must be provided" since the if statement requires both name and team to be defined

Reply:
no, the condition in the if statement is satisfied if both name and team are not defined